src/engine/strat_engine/pool.rs

```diff
@@ -377,12 +377,13 @@ impl StratPool {
     /// This method and destroy() must keep their DM teardown operations
     /// in sync.
     pub fn teardown(self) -> EngineResult<()> {
-        // TODO: any necessary clean up of filesystems
-        if !self.filesystems.is_empty() {
-            return Err(EngineError::Engine(ErrorEnum::Busy,
-                                           format!("May be unsynced files on device.")));
-        }
         let dm = try!(DM::new());
+
+        // Note: Will fail if filesystems are still mounted
```

Comment:
Ok, please get rid of the note and put something like this in the function header comment:

```Precondition: All filesystems belonging to this pool must be unmounted.```